### ISSUE TITLE
Documentation on Snap app structure is missing

### ISSUE THREAD

**Issue by confiks** [jmoenig/Snap#613]
Hi, I'm trying to dig myself into the Snap source, in an attempt to see if it is feasible to use it in an educational project I'm working on. I'm trying to use a subset of your project, most probably only the script area and the execution window. So, considering two ways to dig, I did a depth-first search on definition and execution order, documenting along the way:
- 'contributing to BYOB4.txt' tells me a lot about contributing to the project, but very little about how the project is structured, and what responsibilities individual component/modules (Morphs!) have.
- snap.html lists a lot of Javascript files that push a lot of objects on the global scope, all of them 'Morphs'. The inclusion order might explain how the different components interdepend. Morphic.js is included first, then come GUI dependencies and the GUI, after which the subject matter of Snap is included: blocks and rules and data.
- Morphic.js is beautifully documented, and explains how Morphic works, and how morphs interact with each other. It unfortunately does not tell how the Snap application that is written on top works.
- The HTML page just executes: new IDE_Morph.openIn(new WorldMorph([canvas]))
- WorldMorph really belongs to Morphic, so IDE_Morph must define the Snap application. It does so by first calling buildPanes(), which builds all panes by creating zillion children and grandchildren Morphs. Then it adds itself as a child to the WorldMorph via add() and inherited Node's addChild().

At that point, all parts of the application are set in motion, and it become hard to see how the 'control flow' flows. After studying the code for a small hour I notice that responsibilities for many features are spread out across components. I would for example really like to understand how the state of the execution window is altered by the scripts that are defined in the scripts window.
## tl;dr

Could you please write down what you would tell someone in a quick 5 minutes, about how your project is structured, how the moving parts (of Snap!) interact, and how the program defined by the blocks is executed.
